doc/source/dev/contributor/recommended_development_setup.rst

```diff
@@ -0,0 +1,54 @@
+.. _recommended-development-setup:
+
+=============================
+Recommended development setup
+=============================
```

Comment:
Wasn't sure where to put this, so it's just a separate document linked from CTOC. All of this information is in a more complete setting in the [macOS Quickstart](https://github.com/scipy/scipy/pull/10286/files#diff-23d530dd47e20cba4296e7e42f36ac9c); this might still be needed for Linux/Windows users. It probably should be integrated into those instructions, but I am not modifying those in this PR.